authentication/AuthenticationExample/Utility/CryptoUtils.swift

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import CryptoKit
+
+/// Set of utility APIs for generating cryptographical artifacts.
+enum CryptoUtils {
+  enum NonceGenerationError: Error {
+    case generationFailure(status: OSStatus)
+  }
+
+  static func randomNonceString(length: Int = 32) throws -> String {
+    precondition(length > 0)
```

Comment:
Thanks for adding the precondition.

Reply:
I saw those and thought similarly. Do you happen to know the requirements of the SiwA nonce string? Like, why do we have that particular charset? I was trying to find nonce guidelines for SiwA, but couldn't so I resorted to simplifying the existing implementation. Does it to have to have a length of 32?

Reply:
Based on [this](https://developer.apple.com/documentation/authenticationservices/asauthorizationopenidrequest/3362520-nonce), it looks like we could be more flexible...

Reply:
I was going to suggest the same thing with ChaChaPoly since the project is iOS 13+. I quickly tried it out with the specified character set:
```
import Foundation
import CryptoKit

let charset: [Character] =
Array("0123456789ABCDEFGHIJKLMNOPQRSTUVXYZabcdefghijklmnopqrstuvwxyz-._")

let nonce = ChaChaPoly.Nonce().map { byte in
  // Pick a random character from the set, wrapping around if needed.
  charset[Int(byte) % charset.count]
}

print(String(nonce))
```

That said, I also don't know the requirements on length and allowed chars. I wonder if it would be best to submit as-is with a TODO to look into alternate nonce generation approaches since the PR LGTM otherwise.

Reply:
Good plan @andrewheard, the property we are setting the nonce to mentions that length [may be variable](https://developer.apple.com/documentation/authenticationservices/asauthorizationopenidrequest/3362520-nonce). But I'm not sure about the allowed chars, and agree that clarification on both is best.

I'll (1) post to apple dev forums, (2) link here when I have it, (3) then add a TODO and merge.

Reply:
Post: https://developer.apple.com/forums/thread/728008

Reply:
I'll forgo the TODO as I'll surely update this QS when I learn more :)